logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageRepository.kt

```diff
@@ -187,7 +188,9 @@ class MessageDataSource(
                     // todo: needs some clean up!
                     id = conversationId.toModel(),
                     conversationName = messages.first().conversationName ?: "",
-                    messages = messages.map { message -> messageMapper.fromMessageToLocalNotificationMessage(message) },
+                    messages = messages.sortedBy { message -> message.date }
```

Comment:
Can't we sort in the SQL-level?

Reply:
but still not sure, if we sort them in total, and then filter them based on the conversationId in kotlin, the result would be sorted or not! I'll check that.

Reply:
According to Kotlin's [groupBy](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/group-by.html), it should keep the order it was in:

> The returned map preserves the entry iteration order of the keys produced from the original array.

So sorting them as a whole and then grouping by conversation Id should already result in the correct result.

Just gotta be careful to fetch the newest first, and then we only care about the newest 10.